### PR TITLE
adds `config` field for having mirror fallbacks

### PR DESCRIPTION
**Pull request by scriptnull**
Refer README changes in this PR

Passing tests: https://buildkite.com/hasura/smooth-checkout-buildkite-plugin-integration-tests/builds/5#8f50705d-808c-4e98-848c-d22315d5c49f

### CODE REVIEW

File: plugin.yml [hunk 1]
```diff
@@ -6,8 +6,6 @@ configuration:
   properties:
     skip_checkout:
       type: boolean
-    clone_url:
-      type: string
-    ref:
-      type: string
+    config:
+      type: array
```
Comment (imperfect-fourth) [inline plugin.yml:10]: Have a look at [this](https://stackoverflow.com/questions/36757949/json-schema-definition-for-array-of-objects) to define a more accurate and informative schema

File: hooks/checkout [hunk 2]
```diff
@@ -29,19 +29,68 @@ if ! ssh-keygen -F github.com > /dev/null; then
...
+
+  git clone --no-checkout "$CLONE_URL" && cd "$CLONE_DIR"
+  local EXIT_STATUS=$?
+  if [[ $EXIT_STATUS -ne 0 ]]; then
+    return $EXIT_STATUS
+  fi
+
+  if [ -z "${CHECKOUT_REF}" ]; then
+    echo "Checking out branch: $BUILDKITE_BRANCH with $BUILDKITE_COMMIT"
+    git fetch --force origin "$BUILDKITE_BRANCH" && git checkout --quiet --force "$BUILDKITE_BRANCH" && git reset --hard "$BUILDKITE_COMMIT"
```
Comment (imperfect-fourth) [inline hooks/checkout:58]: let's wrap this line at `&&`s
```bash
git fetch --force origin "$BUILDKITE_BRANCH" && \
  git checkout --quiet --force "$BUILDKITE_BRANCH" && \
  git reset --hard "$BUILDKITE_COMMIT"
```

File: README.md [hunk 3]
```diff
@@ -17,21 +17,33 @@ steps:
...
+    plugins:
+      - hasura/smooth-checkout#v2.0.0:
+          config:
+            - url: git@mirror.git.interal:/path/to/git/mirror
+            - url: git@github.com:<username>/<reponame>.git
+```
 
-Smooth Checkout overcomes this by setting up a workspace directory at `$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID`. The workspace path is made available as `WORKSPACE` environment variable in command section.
+## Setup & Cleanup
+Smooth Checkout setups a workspace directory for your jobs in a non-conflicting fashion. Smooth Checkout overcomes this by setting up a workspace directory at `$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID`. The workspace path is made available as `WORKSPACE` environment variable in command section.
```
Comment (imperfect-fourth) [inline README.md:46]: "Smooth Checkout ~setups~ sets up..."
"Smooth Checkout ~overcomes~ achieves..."

File: README.md [hunk 4]
```diff
@@ -17,21 +17,33 @@ steps:
...
-          clone_url: https://github.com/<username>/<reponame>
-          ref: <ref>
+      - hasura/smooth-checkout#v2.0.0:
+          config:
+            - url: git@github.com:<username>/<reponame>.git
+              ref: <ref>
 ```
-If `clone_url` is not provided, the plugin uses the `BUILDKITE_REPO` env var's value as the target repo. Similarly, if `ref` is not provided the value of `BUILDKITE_BRANCH` env var is used.
+
+If `ref` is not provided the value of `BUILDKITE_BRANCH` and `BUILDKITE_COMMIT` env vars are used.
```
Comment (imperfect-fourth) [inline README.md:26]: value -> values

**Review by imperfect-fourth — APPROVED** [review]
Mainly nitpicks, otherwise looks good

**Comment by scriptnull** [comment]
Passed one more round of testing with the PR comments addressed - https://buildkite.com/hasura/smooth-checkout-buildkite-plugin-integration-tests/builds/6#51fb5c89-8933-4ebf-b914-5df4ee784c97

**Pull request merged**